home_work_converter/Converter.h

```diff
@@ -0,0 +1,80 @@
+//
+// Created by flyon21 on 07.12.24.
+//
+
+#ifndef C_PRO_CONVERTER_H_converter
+#define C_PRO_CONVERTER_H_converter
+
+#include <iostream>
+#include <string>
+#include <stdexcept>
+#include <vector>
+
+enum class ConversionType { Length, Mass, Temperature };
+
+template <ConversionType Type>
+struct ConversionFactor {
+    const char* from;
+    const char* to;
+    double factor;
+    double (*convert)(double) = nullptr;
+};
+
+std::vector<std::string> getAbleUnits(ConversionType type) {
```

Comment:
If you ConversionType type will be const it help you to avoid typo mistake in the feature. For example:
`if (type = ConversionType::Length)`